app/models/shipit/deploy_stats.rb

```diff
@@ -0,0 +1,55 @@
+module Shipit
+  class DeployStats
+    def initialize(deploys)
+      @deploys = deploys
+      @durations = @deploys.map { |d| d.duration.value }.compact
+    end
+
+    def count
+      @deploys.length
+    end
+
+    def average_duration
+      return if @durations.empty?
+      (@durations.sum / @durations.length.to_f)
+    end
+
+    def max_duration
+      @durations.max
+    end
+
+    def min_duration
+      @durations.min
+    end
+
+    def median_duration
+      return if @durations.empty?
+      (sorted_durations[(@durations.length - 1) / 2] + sorted_durations[@durations.length / 2]) / 2.0
+    end
+
+    def success_rate
+      return if @deploys.empty?
+      (@deploys.count(&:success?) / @deploys.length.to_f) * 100
+    end
+
+    def compare(compare_stats)
+      {
+        count: percent_change(count, compare_stats.count),
+        average_duration: percent_change(average_duration, compare_stats.average_duration),
+        median_duration: percent_change(median_duration, compare_stats.median_duration),
+      }
+    end
+
+    protected
+
+    def sorted_durations
+      @sorted ||= @durations.sort
+    end
+
+    def percent_change(new_value, old_value)
```

Comment:
Nitpick: not fan of `percent_change(new_value, old_value)` as a method signature.

I think `percent_change(from, to)` would be better.

Reply:
Updated